test/misc.jl

```diff
@@ -55,3 +55,26 @@ end
         end
     end
 end
+
+@testset "binomlogpdf" begin
+    @testset "identity" begin
+
+        for p in push!(rand(50), [0, 1]...)
+            for n in 0:50
+                for k in push!(collect(Any, -50:50), randn(10)...)
+                    @test binomlogpdf(n, Float32(p), k) ≈ log(pdf(Binomial(n, p), k))
```

Comment:
Isn't this a tautological test, since `Binomial` uses `binomlogpdf` under the hood?

Reply:
So `Distributions.binompdf` has two methods, if `p` is of type `Union{Int64, Float64}`, then it makes a `ccall` to an underlying C library, which, as far as I can tell, is bug free and always returns the correct answer; please see

_src/rmath.jl:70_

However when `p` is of type `Real` but not of type `Union{Int64, Float64}`, then it dispatches on the `binompdf`, found in

`src/distrs/binom.jl:16`

So the above testing code is simply checking that the methods match, under assumption that the method which dispatches on the C code is correct (which I'm pretty sure it is).

Reply:
OK, thanks. Though can you call `binomlogpdf` directly instead of using `log(pdf(Binomial(...), ...))`? `Binomial` is not defined in this package, but in Distributions. Maybe also add a comment explaining that you test the generic Julia definition against Rmath.

Reply:
fixed.